include/picongpu/plugins/radiation/Radiation.kernel

```diff
@@ -56,430 +58,482 @@
-        /// calculate radiated Amplitude
-        /* parallelized in 1 dimensions:
-         * looking direction (theta)
-         * (not anymore data handling)
-         * create shared memory for particle data to reduce global memory calls
-         * every thread in a block loads one particle and every thread runs
-         * through all particles and calculates the radiation for one direction
-         * for all frequencies
+        /**
+         * The radiation kernel calculates for all particles on the device the
+         * emitted radiation for every direction and every frequency.
+         * The parallelization is as follows:
+         *  - There are as many Blocks of threads as there are directions for which
+         *    radiation needs to be calculated. (A block of threads shares
+         *    shared memory)
+         *  - The number of threads per block is equal to the number of cells per
+         *    super cells which is also equal to the number of particles per frame
+         *
+         * The procedure starts with calculating unique ids for the threads and
+         * initializing the shared memory.
+         * Then a loop over all super cells starts.
+         * Every thread loads a particle from that super cell and calculates its
+         * retarded time and its real amplitude (both is dependent of the direction).
+         * For every Particle
+         * exists therefor a unique space within the shared memory.
+         * After that, a thread calculates for a specific frequency the emitted
+         * radiation of all particles.
+         * @param pb
+         * @param radiation
+         * @param globalOffset
+         * @param currentStep
+         * @param mapper
+         * @param freqFkt
+         * @param simBoxSize
          */
+        template<
+            typename ParBox,
+            typename DBox,
+            typename Mapping,
+            typename T_Acc
+        >
+        DINLINE
+        /*__launch_bounds__(256, 4)*/
+        void operator()(
+            T_Acc const & acc,
+            ParBox pb,
+            DBox radiation,
+            DataSpace<simDim> globalOffset,
+            uint32_t currentStep,
+            Mapping mapper,
+            radiation_frequencies::FreqFunctor freqFkt,
+            DataSpace<simDim> simBoxSize
+        ) const
+        {
+            using namespace mappings::threads;
+            namespace po = boost::program_options;
 
-        const int blockSize=pmacc::math::CT::volume<Block>::type::value;
-        // vectorial part of the integrand in the Jackson formula
-        PMACC_SMEM( acc, real_amplitude_s, memory::Array< vector_64, blockSize > );
+            constexpr uint32_t frameSize = pmacc::math::CT::volume< SuperCellSize >::type::value;
+            constexpr uint32_t numWorker = T_numWorkers;
 
-        // retarded time
-        PMACC_SMEM( acc, t_ret_s, memory::Array< picongpu::float_64, blockSize > );
+            using FrameType = typename ParBox::FrameType ;
```

Comment:
QUESTION:
Does `FrameType` replaces `particlesInFrame` from the old code?
Could you  port the comment as well?

Reply:
same one line below